pkg/istio-agent/xds_proxy.go

```diff
@@ -444,6 +449,18 @@ func (p *XdsProxy) handleUpstreamRequest(con *ProxyConnection) {
 				return
 			}
 
+			// override the first xds request node metadata labels
+			// TODO: if we want to regenerate all metadata, use `GetNodeMetaData`
+			if !initialRequestsSent.Load() {
```

Comment:
`initialRequestsSent` is not the same has "this is the first request"; it means we sent the custom Istio types, which occur after LDS. This just happens to work because `req.Node` is nil for all subsequent requests due to `set_node_on_first_message_only`. If `set_node_on_first_message_only=false` was ever used, I think we would set the labels for the first 2 requests, but not subsequent ones which would be incorrect.

I think the more correct and simpler approach is just to always do this if req.Node is non-nil